.github/workflows/workflow.yml

```diff
@@ -70,70 +58,86 @@ jobs:
       - name: Verify dotnet
-        if: runner.os != 'windows'
-        run: __tests__/verify-dotnet.sh 3.1.201 2.2.402
-      - name: Verify dotnet (Windows)
-        if: runner.os == 'windows'
+        shell: pwsh
         run: __tests__/verify-dotnet.ps1 3.1.201 2.2.402
 
-      # Set new cache before 2 digit install
-      - name: Set new tool cache (macOS)
-        if: runner.os == 'macos'
-        run: |
-          echo "DOTNET_INSTALL_DIR=/Users/runner/.dotnet2" >> $GITHUB_ENV
-      - name: Set new tool cache (Ubuntu)
-        if: runner.os == 'linux'
-        run: |
-          echo "DOTNET_INSTALL_DIR=/home/runner/.dotnet2" >> $GITHUB_ENV
-      - name: Set new tool cache (Windows)
-        if: runner.os == 'windows'
-        shell: bash
-        run: |
-          echo "DOTNET_INSTALL_DIR=$LocalAppData\Microsoft\dotnet2" >> $GITHUB_ENV
+  test-setup-without-patch-version:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, windows-latest, macOS-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Clear toolcache
+        shell: pwsh
+        run: __tests__/clear-toolcache.ps1 ${{ runner.os }}
       # 2.0, 3.0, 5.0 needs to be in single quotes to interpret as a string instead of as an integer
-      - name: Setup dotnet '2.0'
+      - name: Setup dotnet '3.1'
         uses: ./
         with:
-          dotnet-version: '2.0'
-
-      # Clear cache before .x version install
-      - name: Set new tool cache (macOS)
-        if: runner.os == 'macos'
-        run: |
-          echo "DOTNET_INSTALL_DIR=/Users/runner/.dotnet3" >> $GITHUB_ENV
-      - name: Set new tool cache (Ubuntu)
-        if: runner.os == 'linux'
-        run: |
-          echo "DOTNET_INSTALL_DIR=/home/runner/.dotnet3" >> $GITHUB_ENV
-      - name: Set new tool cache (Windows)
-        if: runner.os == 'windows'
-        shell: bash
-        run: |
-          echo "DOTNET_INSTALL_DIR=$LocalAppData\Microsoft\dotnet3"  >> $GITHUB_ENV
-      - name: Setup dotnet 2.0.x
+          dotnet-version: '3.1'
+      - name: Setup dotnet '2.2'
+        uses: ./
+        with:
+          dotnet-version: '2.2'
+      - name: Verify dotnet
+        shell: pwsh
+        run: __tests__/verify-dotnet.ps1 3.1 2.2
+  
+  test-setup-latest-patch-version:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04, windows-latest, macOS-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Clear toolcache
+        shell: pwsh
+        run: __tests__/clear-toolcache.ps1 ${{ runner.os }}
+      - name: Setup dotnet 3.1.x
         uses: ./
         with:
-          dotnet-version: 2.0.x
+          dotnet-version: 3.1.x
+      - name: Setup dotnet 2.2.x
```

Comment:
May be we can use matrix to test different versions?

Reply:
The dotnet project that building in tests requires both 2.2 and 3.1 sdks, so, we have to install two versions side-by-side. As I see, we will be installed only one version if we use matrix, am I wrong?